### ISSUE TITLE
[drift_postgres] Query will fail in isolate when Database.connect() is used to init the database

### ISSUE THREAD

**Issue by ynnob** [simolus3/moor#2219]
This bug report is for the experimental implementation of `extras/drift_postgres`.

**Describe the bug**
While working fine when creating the database with the default constructor, the use of the .connect constructor to run the drift instance in an isolate will result in an error while sending a query to the database.

```
ERROR - 2022-12-18 19:51:16.446546
POST /auth/register
Error thrown by handler.

PostgreSQLSeverity.unknown 42601: Syntaxfehler bei Â»;Â« 
DriftCommunication.request                                          package:drift/…/remote/communication.dart:109
_BaseExecutor._runRequest                                            package:drift/…/remote/client_impl.dart:85
_BaseExecutor.runSelect                                                 package:drift/…/remote/client_impl.dart:118
SimpleSelectStatement._getRaw.<fn>                           package:drift/…/select/select.dart:68
DatabaseConnectionUser.doWhenOpened.<fn>          package:drift/…/api/connection_user.dart:160
```

![drift_error2](https://user-images.githubusercontent.com/34171562/208314641-307d9ac2-8ad3-4865-9287-c475f51ddbe9.png)

Method that produces the error:
```
    // Check if user already exists
    final dbUser = await (db.select(db.users)
          ..where((tbl) => tbl.email.equals(requestedMail)))
        .getSingleOrNull();
```
Query that gets send to the server and produces the error:
`SELECT * FROM "users" WHERE "email" = ?;`

![drift_error](https://user-images.githubusercontent.com/34171562/208314535-cd61cc50-ca1c-4a06-b0a2-84ba05524324.png)

Database class:
```
@DriftDatabase(tables: [Users, TokenFamilies, UsedTokens])
class PostgresDatabase extends _$PostgresDatabase {
  PostgresDatabase() : super(openConnection());

  PostgresDatabase.connect(DatabaseConnection connection)
      : super.connect(connection);
}

```

Isolate Handler:
```
class PostgresServer {
  static Future<DriftIsolate> run() async {
    return await DriftIsolate.spawn(backgroundConnection);
  }
}

PgDatabase openConnection() {
  final pgConnection = PostgreSQLConnection('127.0.0.1', 5432, 'postgres',
      username: 'postgres', password: 'postgres');
  return PgDatabase(pgConnection);
}

DatabaseConnection backgroundConnection() {
  return DatabaseConnection(openConnection());
}
```

The database is used in a shelf dart server enviroment and a connection is opened like this:
```
void main(List<String> args) async {
  final dbIsolate = await startDatabaseServer();
  final dbConnection = await dbIsolate.connect();
  final db = PostgresDatabase.connect(dbConnection);
}

Future<DriftIsolate> startDatabaseServer() async => await PostgresServer.run();
```

As mentiones above this will result in an error when a query gets send to the database. If i use `final db = PostgresDatabase();` the query gets executed as expected.

I know that drift_postgres is only an experimental feature but i hope you still accept bug reports for it.

**Comment by simolus3** [comment]
Thanks for the report! This should be fixed in 18c6139eb35c572be8effadc24f7f9437ecb0b39.

Basically, the problem is that drift generates queries for sqlite3 unless the underlying database client tells it that it's supposed to use something else. Unfortunately, that information got lost in the remote isolate, so here drift is generating a query with a wrong variable format.

But note that isolates are not necessarily required with drift postgres since the `postgres` package is asynchronous already.

**Comment by ynnob** [comment]
Thanks @simolus3 !

Jeah maybe i dont need to use an isolate. Not really sure what the best approach is.
My Server spins up multiple isolates depending on how many cpu cores the system has. i thought i could maybe increase performance if i spin up one isolate that only manages the drift instance and the other isolates that listen for endpoint connections could connect through the DriftIsolate.

I guess i need to test to find out what the most performant solutuion is.
